test/extensions/filters/http/dynamic_forward_proxy/proxy_filter_integration_test.cc

```diff
@@ -142,5 +182,31 @@ TEST_P(ProxyFilterIntegrationTest, DNSCacheHostOverflow) {
   EXPECT_EQ(1, test_server_->counter("dns_cache.foo.host_overflow")->value());
 }
 
+// Verify that upstream TLS works with auto verification for SAN as well as auto setting SNI.
```

Comment:
I assume we'll do mismatch failure as part of your test TODOs?

Reply:
I will work on adding a test for this, though I will likely have to add another upstream cert which doesn't have the "localhost" SAN.